### PR TITLE
Fix issue #64732

### LINKED ISSUE #64732: Incorrect error message for byte literal with single quotes
When creating a byte literal with single quotes, like `b'hello\0'`, The error message says:
```
error: character literal may only contain one codepoint
   --> src/wire_protocol/operations.rs:101:37
    |
101 |         assert_eq!(Bson::Binary(4, b'12345').byte_length().unwrap(), 10);
    |                                     ^^^^^^^
help: if you meant to write a `str` literal, use double quotes
    |
101 |         assert_eq!(Bson::Binary(4, b"12345").byte_length().unwrap(), 10);
    |                                     ^^^^^^^
```

But following the directions would not give us a `str` literal, it would give a `[u8]` literal.

### PR DESCRIPTION
**Pull request by onehr**
Based on issue #64732, when creating a byte literal with single quotes,
the suggestion message would indicate that you meant to write a `str` literal,
but we actually meant to write a byte string literal.

So I changed the unescape_error_reporting.rs to decide whether to print out
"if you meant to write a `str` literal, use double quotes",
or "if you meant to write a byte string literal, use double quotes".

Fixes #64732.

### CODE REVIEW

File: src/libsyntax/parse/unescape_error_reporting.rs [hunk 1]
```diff
@@ -47,14 +47,20 @@ pub(crate) fn emit_unescape_error(
                 .emit();
         }
         EscapeError::MoreThanOneChar => {
+            let sug_msg = if mode.is_bytes() {
```
Comment (varkor) [inline src/libsyntax/parse/unescape_error_reporting.rs]: We typically just call this `msg`.

**Comment by varkor** [comment]
@onehr: thanks, this looks good! Could you add a test somewhere in `src/test/ui` to make sure it's working correctly? After you add the test, you can run `x.py test src/test/ui/<TEST> --bless` to generate the output file and make sure it's passing. [The rustc guide has more detailed help.](https://rust-lang.github.io/rustc-guide/tests/adding.html)

**Comment by onehr** [comment]
@varkor Thanks for your help? So do I need to put the test into any specific file or it's ok to create a new one and name it.

**Comment by varkor** [comment]
You can create a new one. If you can come up with a relevant name, you can put it under `src/test/ui` directly, or if not, you can just create a new test at `src/test/ui/issues/issue-64732.rs`.

**Comment by onehr** [comment]
Hi @varkor, I have changed the code based on your review and added the test case for it, can you give it a check?

**Comment by varkor** [comment]
@onehr: that's perfect, thanks for fixing this issue!

@bors r+ rollup

**Pull request merged**